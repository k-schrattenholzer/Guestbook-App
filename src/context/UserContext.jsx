// Create the Context
// Create the UserProvider component
// Create custom hook

 // `context` will be undefined when using this custom hook
  // outside of the provider, so we throw an error 

  // Export the provider component and the custom hook