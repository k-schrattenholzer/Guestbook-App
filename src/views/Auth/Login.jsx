import { useHistory, useLocation } from "react-router-dom";
import { useState } from 'react';
import { useUser } from "../../context/UserContext.jsx";
import useForm from '../../hooks/useForm.js'


export default function Login() {
  const history = useHistory();
  const location = useLocation();
  const auth = useUser();

  const { formState, handleFormChange } = useForm({ username: '', password: ''});
  console.log(formState)
  const [error, setError] = useState(null);

  const { from } = location.state || { from: { pathname: '/' } };

  const handleLogin = (e) => {
    e.preventDefault();
    const userLoggedIn = auth.login(formState.username, formState.password);

    return !loginWasSuccessful ?
    setError("login unsuccessful bb, try again?") : history.replace('/')  
  }

  return(
    <fieldset className="w-1/4 border p-4">
  <legend>Sign In</legend>
  <form 
    className="grid grid-cols-[1fr_2fr] grid-rows-3"
    onSubmit={handleLogin}
    >
    <label htmlFor="username" className="self-center text-right">
      Username
    </label>
    <input
      id="username"
      type="text"
      name="username"
      value={formState.username}
      className="border p-2 m-3"
      required
      onChange={handleFormChange}
    />
    <label htmlFor="password" className="self-center text-right">
      Password
    </label>
    <input
      id="password"
      type="password"
      name="password"
      value={formState.password}
      className="border p-2 m-3"
      required
      onChange={handleFormChange}
    />
    <button
      type="submit"
      className="h-12 self-center bg-black text-white p-1 font-bold col-span-2"
    >
      Sign In
    </button>
  </form>
  <p className="text-center text-red-500 font-bold">Error message goes here</p>
</fieldset>
  );
}