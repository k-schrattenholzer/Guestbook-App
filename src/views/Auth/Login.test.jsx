import {
  fireEvent,
  render,
  screen,
  waitFor
} from '@testing-library/react';
import { MemoryRouter, Route, Switch } from 'react-router-dom';
import { UserProvider } from '../../context/UserContext.jsx';
import Login from './Login.jsx';


const mockLogout = jest.fn();

jest.mock('../../hooks/useAuth', () => {
  return {
    useAuth: function () {
      return {
        user: { username: 'k@k.com' },
        logout: mockLogout,
      }
    }
  }
});

it('should render the login form', () => {
  const { container } = render (
    <UserProvider>
      <MemoryRouter>
        <Login />
      </MemoryRouter>
    </UserProvider>
  )
  expect(container).toMatchSnapshot();
})

it('should allow guest to sign in', async () => {
  const { container } = render(
    <UserProvider>
      <MemoryRouter>
        <Switch>
          <Route>

          </Route>
        </Switch>
      </MemoryRouter>
    </UserProvider>
  )
})