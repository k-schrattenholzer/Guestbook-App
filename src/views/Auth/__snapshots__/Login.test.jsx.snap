// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the login form 1`] = `
<div>
  <fieldset
    class="w-1/4 border p-4"
  >
    <legend>
      Sign In
    </legend>
    <form
      class="grid grid-cols-[1fr_2fr] grid-rows-3"
    >
      <label
        class="self-center text-right"
        for="username"
      >
        Username
      </label>
      <input
        class="border p-2 m-3"
        id="username"
        name="username"
        required=""
        type="text"
        value=""
      />
      <label
        class="self-center text-right"
        for="password"
      >
        Password
      </label>
      <input
        class="border p-2 m-3"
        id="password"
        name="password"
        required=""
        type="password"
        value=""
      />
      <button
        class="h-12 self-center bg-black text-white p-1 font-bold col-span-2"
        type="submit"
      >
        Sign In
      </button>
    </form>
    <p
      class="text-center text-red-500 font-bold"
    />
  </fieldset>
</div>
`;
